docs/architecture/adr-002-ibc-relayer.md

```diff
@@ -0,0 +1,644 @@
+
+#### Keyring 
+The relay process must have access to its accounts with tokens on all destination chains, with sufficient balance to pay for transaction fees. Account key information must be stored and managed securely. A keyring implementation is required for CRUD key operations. 
+[Keyring-Rust] Investigation in existing Rust implementations is needed. (ex: [hwchen-keyring](https://github.com/hwchen/keyring-rs))
+
+### Chain Transactions and Signing
+The relayer must create chain specific signed transactions. 
+[Cosmos-Tx-Rust] For the first release Cosmos-SDK transaction signing is required. One possible implementation is [iqlusion's sdtx crate](https://github.com/iqlusioninc/crates/tree/develop/stdtx)
+
+#### Implementation of IBC "routing module"
+The default IBC handler uses a receiver call pattern, where modules must individually call the IBC handler in order to bind to
+ports, start handshakes, accept handshakes, send and receive packets, etc. While this provides flexibility for modules, it imposes extra work on the part of the relayer processes that now needs to track the state of multiple modules. The IBC specification describes an IBC “routing module” to route packets, and simplify the task of relayers. This routing module accepts external datagrams and calls into the IBC handler to deal with handshakes and packet relay. The routing module keeps a lookup table of modules, which it can use to look up and call a module when a packet is received, so that external relayers need only ever relay packets to the routing module.
+[IBC-Routing-Module-Go] Initial version of the relayer assumes that chains implement the "routing module"
+
+#### Batching
+The relayer may batch transactions if supported by destination chain and allowed by configuration. In this case the relayer  can bundle multiple datagrams into a single transaction and amortise any overhead costs (e.g. signature checks for fee payment).
+Initial version of the relayer assumes batching is supported by all chains. An option may be later included in the configuration file.
+
+## Relayer Requirements
+
+A correct relayer MUST:
+
+- **[R-config-start]** Read, parse, validate a configuration file upon start and configure itself for the specifed chains and paths
+- **[R-transport]** Have access to the networking protocols (e.g. TCP/IP, UDP/IP, or QUIC/IP) and physical transport, required to read the state of one blockchain/ machine and submit data to another
+- **[R-provider]** Maintain transport connections to at least one full node per chain
+- **[R-query]** Query IBC data on source and destination chains
+- **[R-light-client]** Run light clients for source chains and 
+- **[R-IBC-client]** create and update IBC clients on destination chains 
+- **[R-accounts]** Own accounts on destination chains with sufficient balance to pay for transaction fees
+- **[R-transact]** Create, sign and forward IBC datagram transactions
```

Comment:
Might be good idea defining IBC datagram transaction (a transaction that contains IBC packets (we use datagram term sometimes).

Reply:
attempted a fix here